docs/develop/howtos/override_components.md

```diff
@@ -0,0 +1,121 @@
+# How to override a react component
+
+This documentation is targeted to developers that want to customize a specific react component in an instance.
+
+## Overridable
+
+It is possible to override certain react components in Invenio RDM, by using the library [`react-overridable`](https://github.com/indico/react-overridable). This feature is available to instance developers to customize the look and feel of the react application (e.g. deposit form).
+
+Despite the library allowing multiple ways of overriding a component, for this guide we will implement a new component to replace the default one.
+
+## Steps
+
+- Make sure that the component you want to override is implemented in the same level as an `override.js` file. For example, for the deposit form, `invenio-app-rdm` has the following structure:
+
+```terminal
+├── invenio_app_rdm
+│   ├── assets
+│   │   ├── semantic-ui
+│   │   |   ├── js
+│   │   |   |   ├── invenio_app_rdm
+│   │   |   |   |   ├── deposit
+│   │   |   |   |   |   ├── override.js
+│   │   |   |   |   |   ├── RDMDepositForm.js
+```
+
+We will override the file `override.js`.
+
+- Find a react component to override. For this example, we are going to override the deposit form '`Metadata-only record`' checkbox.
+
+!['`Metadata-only record`' checkbox](./img/metadata_only_checkbox.png)
+
+!!! info "You must know the component id beforehand"
+    In order to override a component, the exact overridable id must be retrieved beforehand.
+    For now, you must retrieve the component id directly from the module's code.
+
+The react component's overridable id for the '`Metadata-only record`' checkbox is `ReactInvenioDeposit.MetadataAccess.layout`.
+
+For this example, we will replace the '`Metadata-only record`' checkbox with another component, like a [Toggle](https://react.semantic-ui.com/modules/checkbox/#types-toggle) component.
+
+- Create a file named `override.js` with the same path as the one defined by the module (e.g. `<instance_name>/assets/js/invenio_app_rdm/deposit/override.js`). Note that `semantic-ui` was ommited.
+
+- Create a file (e.g. `MetadataToggle.js`) to implement your component, under `<instance_name>/assets/js/components`. If the folder `components` does not exist, you can create it.
```

Comment:
Suggestion: The new components do not have a requirement on where they have to be, I would maybe phrase it as a recommendation to keep/create a components folder, you could even add another level after it, to group the components like `.../components/deposit`.

Reply:
What do you mean with "the linking problem"? We override by `id`, not by replacing the file.